src/components/contextual/pages/vebal/cross-chain-boost/SyncNetworkAction.vue

```diff
@@ -0,0 +1,177 @@
+<script setup lang="ts">
+import configs, { Network } from '@/lib/config';
+import { buildNetworkIconURL } from '@/lib/utils/urls';
+import { TransactionActionInfo } from '@/types/transactions';
+import { useI18n } from 'vue-i18n';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { useCrossChainSync } from '@/providers/cross-chain-sync.provider';
+import useTransactions from '@/composables/useTransactions';
+import useEthers from '@/composables/useEthers';
+
+/**
+ * TYPES
+ */
+type Props = {
+  chosenNetworks: Set<Network>;
+  activeTabIdx: number;
+  veBalBalance: string;
+};
+
+/**
+ * PROPS & EMITS
+ */
+const props = defineProps<Props>();
+const emit = defineEmits(['update:activeTabIdx']);
+
+/**
+ * COMPOSABLES
+ */
+const { t } = useI18n();
+const { addTransaction } = useTransactions();
+const { txListener } = useEthers();
+const {
+  l2VeBalBalances,
+  sync,
+  setTempSyncingNetworks,
+  tempSyncingNetworks,
+  setSyncTxHashes,
+} = useCrossChainSync();
+
+/**
+ * STATE
+ */
+const currentActionIndex = ref(0);
+
+/**
+ * METHODS
+ */
+async function handleTransaction(
+  tx: TransactionResponse,
+  network: Network
+): Promise<void> {
+  addTransaction({
+    id: tx.hash,
+    type: 'tx',
+    action: 'sync',
+    summary: `Sync veBAL to ${configs[network].chainName} network`,
+  });
+
+  txListener(tx, {
+    onTxConfirmed: () => {
+      setSyncTxHashes(network, tx.hash);
+    },
+    onTxFailed: () => {
+      //
```

Comment:
We should display an error message in this case, right? Could you check what is done in the join/exit flows here and see what we might need to add.

Reply:
not sure if we need to add smth here. error message is handled in `BalActionSteps` component and in other flows here `confirming` field in state is set to false. e.g https://github.com/balancer/frontend-v2/blob/7cdfe61e43d37b19d6a0e5f8570256ac1f09867f/src/components/forms/pool_actions/WithdrawForm/components/WithdrawPreviewModal/components/WithdrawActions.vue#L135